anta/tests/routing/bgp.py

```diff
@@ -447,3 +493,82 @@ def test(self) -> None:
 
         if failures:
             self.result.is_failure(f"Failures: {list(failures.values())}")
+
+
+class VerifyBGPExchangedRoutes(AntaTest):
+    """
+    Verifies if the BGP peers have correctly advertised and received routes.
+    The route type should be 'valid' and 'active' for a specified VRF.
+
+    Expected results:
+        * success: If the BGP peers have correctly advertised and received routes of type 'valid' and 'active' for a specified VRF.
+        * failure: If a BGP peer is not found, the expected advertised/received routes are not found, or the routes are not 'valid' or 'active'.
+    """
+
+    name = "VerifyBGPExchangedRoutes"
+    description = "Verifies if BGP peers have correctly advertised/received routes with type as valid and active for a specified VRF."
+    categories = ["routing", "bgp"]
+    commands = [
+        AntaTemplate(template="show bgp neighbors {peer} advertised-routes vrf {vrf}"),
+        AntaTemplate(template="show bgp neighbors {peer} routes vrf {vrf}"),
+    ]
+
+    class Input(AntaTest.Input):
+        """
+        Input parameters of the testcase.
+        """
+
+        bgp_peers: List[BgpNeighbors]
+        """List of BGP peers"""
+
+        class BgpNeighbors(BaseModel):
+            """
+            This class defines the details of a BGP peer.
+            """
+
+            peer_address: IPv4Address
+            """IPv4 address of a BGP peer"""
+            vrf: str = "default"
+            """Optional VRF for BGP peer. If not provided, it defaults to `default`."""
+            advertised_routes: List[IPv4Network]
+            """List of advertised routes of a BGP peer."""
+            received_routes: List[IPv4Network]
+            """List of received routes of a BGP peer."""
+
+    def render(self, template: AntaTemplate) -> list[AntaCommand]:
+        """Renders the template with the provided inputs. Returns a list of commands to be executed."""
+
+        return [
+            template.render(peer=bgp_peer.peer_address, vrf=bgp_peer.vrf, advertised_routes=bgp_peer.advertised_routes, received_routes=bgp_peer.received_routes)
+            for bgp_peer in self.inputs.bgp_peers
+        ]
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        failures: dict[str, dict[str, Any]] = {}
+
+        # Iterating over command output for different peers
+        for command in self.instance_commands:
+            peer = str(command.params.get("peer", ""))
+            vrf = command.params.get("vrf", "")
+            advertised_routes = command.params.get("advertised_routes", [])
+            received_routes = command.params.get("received_routes", [])
```

Comment:
As discussed, in recent ANTA refactor, it looks like `command.params` is properly typed as `Dict[str, Any]` so we can simply do a dict lookup without mypy complaining. No need to bother with `.get()` or `cast` anymore.